sales_team_security/models/res_partner.py

```diff
@@ -40,3 +41,54 @@ def _onchange_parent_id_sales_team_security(self):
         """
         if self.parent_id and self.parent_id.team_id and not self.team_id:
             self.team_id = self.parent_id.team_id.id
+
+    @api.depends('company_type', 'child_ids', 'child_ids.user_id')
```

Comment:
Depends is only for computed field, so this doesn't add anything.

Reply:
Changed it to `commercial_partner_id`